src/main/java/com/google/sps/servlets/MealServlet.java

```diff
@@ -85,13 +123,36 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
         return;
     }
 
+    private Entity createMealEntity(Meal meal) {
+        Entity mealEntity = new Entity("Meal");
+        mealEntity.setProperty("id", meal.getId());
+        mealEntity.setProperty("title", meal.getTitle());
+        mealEntity.setProperty("description", meal.getDescription());
+        mealEntity.setProperty("ingredients", meal.getIngredients());
+        mealEntity.setProperty("type", meal.getType());
+ 
+        return mealEntity;
+    }
+
+    private void getMealList(HttpServletRequest request, HttpServletResponse response) throws IOException {
+        Query query = new Query("Meal").addSort("id", SortDirection.DESCENDING);
```

Comment:
I suppose it is better to sort entities by `title`, it is more common for user.
Or first by `type`, then by `title`.

Reply:
In general that's true, but let's not forget that this is not exactly what the user will see.
This is an API method, which is kind expected to be used by advanced users (scripts or developers, basically). It may make sense to return the data in a format that makes sense to a developer, rather than a regular user.

Thinking about the end user can be implemented by the frontend code. The JS script can simply sort the results while rendering.

Or, to put it differently - if we are convinced that sorting items by name makes sense to the end user, we should make the frontend code sort meals, regardless of whether the backend already sorts them or not. If the frontend is sure that this is the right approach, it should just do it, rather than depend on the backend.